src/KeyVault/KeyVault/Models/KeyVaultManagementCmdletBase.cs

```diff
@@ -213,14 +203,7 @@ protected string GetCurrentUsersObjectId()
             string objectId = null;
             if (DefaultContext.Account.Type == AzureAccount.AccountType.User)
             {
-// TODO: Remove IfDef
-#if NETSTANDARD
                 objectId = ActiveDirectoryClient.GetObjectId(new ADObjectFilterOptions {UPN = DefaultContext.Account.Id}).ToString();
-#else
-                var userFetcher = ActiveDirectoryClient.Me.ToUser();
```

Comment:
ActiveDirectoryClient.Me.ToUser() is more secure than DefaultContext.Account.Id.  If I recall correctly, DefaultContext.Account.Id comes from a non-secure, display only return value from ADAL.  It's not meant to be used as a secure id.  ActiveDirectoryClient.Me.ToUser() actually goes to AAD and gets the user object of the currently signed in user.  Since the id is used to create a default access policy for the vault it should be as secure as possible.  Unless the source of DefaultContext.Account.Id has changed to be AAD Graph itself, you need to find a way of getting the object id that in the same way as ActiveDirectoryClient.me.ToUser().

Reply:
I opened a new issue here: https://github.com/Azure/azure-powershell/issues/8440 to look into this edge case.  I will open a new PR with just the fix for the Graph token issue, and will leave the ifdefs for future reference.